src/miner.cpp

```diff
@@ -219,6 +219,31 @@ std::unique_ptr<CBlockTemplate> BlockAssembler::CreateNewBlock(const CScript& sc
     return std::move(pblocktemplate);
 }
 
+std::unique_ptr<CBlockTemplate> BlockAssembler::PickTransactions()
```

Comment:
In essence, this method just returns a minimal block template, so it looks like the name doesn't fully reflect the logic.
Also, it could be great to use this method inside of `BlockAssembler::CreateNewBlock` - it would remove the duplicated code and makes the original method more eloquent.

Reply:
> it could be great to use this method inside of `BlockAssembler::CreateNewBlock`

Although it's possible it seems to be not so straightforward because of coinbase transaction and mutexes.